### PR TITLE
add [[Log]] like operations

### PR DESCRIPTION
**Pull request by sperlingxx**
## What changes were proposed in this pull request?

add Log/Log1p/LogN/LogN1p operations and logSpec

## How was this patch tested?

unit tests

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Log1p.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import scala.reflect.ClassTag
+
+class Log1p[T: ClassTag, D: ClassTag](implicit ev: TensorNumeric[T], ev2: TensorNumeric[D])
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Log1p.scala]: is this duplicate with **com.intel.analytics.bigdl.nn.tf.Log1p**?

It's a useful layer. Maybe we should move that class to nn package.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Log.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import scala.reflect.ClassTag
+
+class Log[T: ClassTag, D: ClassTag](implicit ev: TensorNumeric[T], ev2: TensorNumeric[D])
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Log.scala]: Is it duplicate with **com.intel.analytics.bigdl.nn.Log**? Maybe append a generic type D to that layer?

Comment (tosky001) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Log.scala]: the `nn.ops.Log` only contains **forward** function for Data Preprocessing. there is another ops **Exp** which are also defined  as `nn.ops.Exp` and `nn.Exp`. So, we have always had a confusion: what is  the specific distinguish between `nn.ops.*` and `nn.*`.  Moreover, how to deal with the common Components like **Log** and **Exp** etc.

Comment (yiheng) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Log.scala]: nn.ops.* cannot do backpropagation, so cannot be used to build the model(e.g. add to sequential and static graph).

Currently, we only use nn.ops.* when load and run tensorflow models. Use operation for data preprocessing is an interesting idea. But we may need to improve the logic in containers. We're open for discussion about this.

Comment (yiheng) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Log.scala]: If the logic of operation Log and layer Log are totally same, I think we can use the layer Log in the place of operation Log, right?

**Pull request closed without merging**